src/components/inputs/UserServiceUnitSelectInput.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import React, {Component} from 'react';
+import flowRight from 'lodash/flowRight';
+import {connect} from 'react-redux';
+import {reduxForm} from 'redux-form';
+import Select from 'react-select';
+
+import FormFieldLabel from '$components/form/FormFieldLabel';
+import DropdownIndicator from '$components/inputs/DropdownIndicator';
+import LoadingIndicator from '$components/inputs/SelectLoadingIndicator';
+import {FormNames} from '$src/enums';
+import {setUserActiveServiceUnit} from '$src/usersPermissions/actions';
+
+import type {UserServiceUnit, UserServiceUnits} from '$src/usersPermissions/types';
+
+type Props = {
+  userServiceUnits: UserServiceUnits,
+  userActiveServiceUnit: UserServiceUnit,
+  setUserActiveServiceUnit: Function,
+}
+
+class UserServiceUnitSelectInput extends Component<Props, State> {
+  handleChange = (val: any) => {
+    const {
+      setUserActiveServiceUnit,
+      userServiceUnits,
+    } = this.props;
+
+    if (val) {
+      const {value} = val;
```

Comment:
If `val` is object and it has (or should have) value attribute, then code could be simplified with optional chaining aka `val?.value`.